### PR TITLE
Remove MBEDTLS_xxx_ALT interface for crypto mechanisms

### PR DESCRIPTION
**Pull request by tom-daubney-arm**
## Description

This PR aims to remove the support for the _ALT interface for cryptographic mechanisms.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [ ] **changelog** TODO
- [x] **3.6 backport** not required
- [x] **2.28 backport** not required
- [x] **tests** existing tests suffice

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

File: include/mbedtls/check_config.h [hunk 1]
```diff
@@ -768,8 +768,8 @@
 #if !defined(MBEDTLS_SHA512_C)
 #error "MBEDTLS_SHA512_USE_A64_CRYPTO_* defined without MBEDTLS_SHA512_C"
 #endif
-#if defined(MBEDTLS_SHA512_ALT) || defined(MBEDTLS_SHA512_PROCESS_ALT)
-#error "MBEDTLS_SHA512_*ALT can't be used with MBEDTLS_SHA512_USE_A64_CRYPTO_*"
+#if defined(MBEDTLS_SHA512_PROCESS_ALT)
+#error "MBEDTLS_SHA512_PROCESS_ALT can't be used with MBEDTLS_SHA512_USE_A64_CRYPTO_*"
```
Comment (tom-cosgrove-arm) [inline include/mbedtls/check_config.h:772]: Should `MBEDTLS_SHA512_PROCESS_ALT` still exist?

Comment (tom-daubney-arm) [reply include/mbedtls/check_config.h:772]: No it shouldn't. This is the removal I mentioned to you where I am going to get this option, and the remaining one like it, in one commit at the end. I only discovered this type of config option when I was halfway through my initial removals.

Comment (tom-daubney-arm) [reply include/mbedtls/check_config.h:772]: Still got probably 5 or so commits to add before I consider this PR done to a first approx.

**Comment by tom-daubney-arm** [comment]
Note that ABI-API checks are failing but this is expected. I have moved some previously public functions to private - thus affecting the API - and would think that the various removals have affected the ABI as well.

**Comment by tom-daubney-arm** [comment]
PLEASE DO NOT REVIEW THIS PR.

This PR is being split into a series to make review less daunting. Once all the sub-PRs are up, this PR will be closed.

The first sub-PR is here: #9259

**Comment by tom-cosgrove-arm** [comment]
@tom-daubney-arm I've removed the `needs-review` and `needs-reviewer` labels, so no-one comes at this to review it, and removed the `high-priority` and added a `DO-NOT-MERGE` for good measure :). How would you feel about making this PR a draft in addition? (I'm just thinking of the reporting statistics!)

**Comment by tom-daubney-arm** [comment]
> @tom-daubney-arm I've removed the `needs-review` and `needs-reviewer` labels, so no-one comes at this to review it, and removed the `high-priority` and added a `DO-NOT-MERGE` for good measure :). How would you feel about making this PR a draft in addition? (I'm just thinking of the reporting statistics!)

Happy to make it draft but it is going to be closed within the next working day.

**Comment by tom-daubney-arm** [comment]
Closing now that the final PR of the series is up: https://github.com/Mbed-TLS/mbedtls/pull/9402.

**Pull request closed without merging**